xarray/namedarray/utils.py

```diff
@@ -195,6 +198,31 @@ def either_dict_or_kwargs(
     return pos_kwargs
 
 
+def fake_target_chunksize(
+    data: Variable | T_ChunkedArray,
+    target_chunksize: int,
+) -> tuple[int, np.dtype[Any]]:
+    """
+    Naughty trick - let's get the ratio of our cftime_nbytes, and then compute
+    the ratio of that size to a np.float64. Then we can just adjust our target_chunksize
+    and use the default dask chunking algorithm to get a reasonable chunk size.
+    """
+    from xarray.core.formatting import first_n_items
+
+    output_dtype = np.dtype(np.float64)
+
+    if data.dtype == object:
+        nbytes_approx: int = sys.getsizeof(first_n_items(data, 1))  # type: ignore[no-untyped-call]
+    else:
+        nbytes_approx = data.dtype.itemsize
```

Comment:
```suggestion
        return limit, var.dtype
```

Otherwise we're doing the rescaling for all variables no?